nilearn/datasets/struct.py

```diff
@@ -562,3 +565,40 @@ def _fetch_surf_fsaverage5_sphere(data_dir=None):
 
     result['description'] = str(_get_dataset_descr('fsaverage5_sphere'))
     return Bunch(**result)
+
+
+def _fetch_surf_fsaverage6(data_dir=None):
+    """Helper function to ship fsaverage5 spherical meshes.
+
+    These meshes can be used for visualization purposes, but also to run
+    cortical surface-based searchlight decoding.
+
+    The source of the data is downloaded from OSF.
+
+    """
+    dataset_dir = _get_dataset_dir('fsaverage6')
+    url = 'https://osf.io/3s4r2/download'
+    opts = {'uncompress': True}
+    names = [
+        "{}.{}".format(hemi, part)
+        for hemi in ["lh", "rh"]
+        for part in ["curl", "inflated", "pial", "white"]
+    ]
+    filenames = [
+        (os.path.join(
+            'tpl-fsaverage',
+            'fsaverage6',
+            'surf',
```

Comment:
I'm wondering why you build such a complex folder system here. I think for fsaverage and fsaverage5 the files are directly under `fsaverage` and `fsaverage5` respectively.